functions/proxy.fish

```diff
@@ -5,21 +5,25 @@ function proxy -d "Setup proxy environment variables"
     return
   end
 
-  # Get user & password
-  set -l user $proxy_user
-  if not set -q proxy_user
-    read -p "echo -n 'Proxy User: '" user
-  end
-  # Hacky way to read password in fish
-  echo -n 'Proxy Password: '
-  stty -echo
-  head -n 1 | read -l pass
-  stty echo
-  echo
-  # URL encode password
-  set -l chars (echo $pass | sed -E -e 's/./\n\\0/g;/^$/d;s/\n//')
-  printf '%%%02x' "'"$chars"'" | read -l encpass
+  if eval $proxy_auth
```

Comment:
I don't like the use of eval here, we are effectively executing whatever the user puts in proxy_auth, which is dangerous.

This would be more idomatic of fish code I've seen.   Quoting the variable is necessary, otherwise test errors out if not it's set:

    if test "$proxy_auth" = "true"